packages/assets-controllers/src/NftController.ts

```diff
@@ -352,12 +353,22 @@ export class NftController extends BaseController<NftConfig, NftState> {
     contractAddress: string,
     tokenId: string,
   ): Promise<NftMetadata> {
-    const { ipfsGateway, useIPFSSubdomains } = this.config;
+    const { ipfsGateway, useIPFSSubdomains, isIpfsGatewayEnabled } =
+      this.config;
     const result = await this.getNftURIAndStandard(contractAddress, tokenId);
     let tokenURI = result[0];
     const standard = result[1];
 
     if (tokenURI.startsWith('ipfs://')) {
+      if (!isIpfsGatewayEnabled) {
+        return {
+          image: null,
+          name: null,
+          description: null,
+          standard: standard || null,
+          favorite: false,
+        };
+      }
       tokenURI = getFormattedIpfsUrl(ipfsGateway, tokenURI, useIPFSSubdomains);
     }
```

Comment:
Nit: With this new change, this `if` branch is doing multiple things, either returning early or not returning early and changing the `tokenURI` variable. Considering that this function is quite large already, it might be easier to parse this code if the early return were split out. Something like:

``` suggestion
    const hasIpfsTokenURI = tokenURI.startsWith('ipfs://');

    if (hasIpfsTokenURI && !isIpfsGatewayEnabled) {
      return {
        image: null,
        name: null,
        description: null,
        standard: standard || null,
        favorite: false,
      };
    }

    if (hasIpfsTokenURI) {
      tokenURI = getFormattedIpfsUrl(ipfsGateway, tokenURI, useIPFSSubdomains);
    }
```

Reply:
Love this addressing it now!